### ISSUE TITLE
Trouble opening ipynb file with a # in the file name

### ISSUE THREAD

**Issue by sadukie** [microsoft/AzureNotebooks#386]
Please describe your issue as completely as possible.

#### Which site did you access us from?
    [x] https://notebooks.azure.com
    [] https://studio.azureml.net

#### Describe the issue you are encountering

Uploaded the Jupyter notebook from my machine (running from Windows 10).  Tried to access it while on Wi-Fi in Hong Kong, and getting the following error:
https://notebooks.azure.com/Error/NotFound?404;https:%2F%2FANS-Prod-USSouth:80%2FrehydrationRedirect.txt#%20Devs.ipynb

Problem occurs when I click Run.  However, this is accessible via JupyterLab.

I was hoping to share this with a user group here in Hong Kong on June 13.  I can just demo with Jupyter Notebooks if I have to.

#### Which language/kernel were you using when you experienced this issue?
    [] Python 2.7 
    [] Python 3.5
    [x] Python 3.6 
    [] R 
    [] F#
    [] Terminal

#### What time (please include time zone) did this occur?
11:17am Hong Kong time - Tuesday, June 5, 2018
#### Which Web Browser(s) did you see the error in?
    [x] Chrome
    [] Firefox
    [] Internet Explorer / Edge (Please note this is not supported)
    [] Other (Please Specify)

#### Which Operating System are you using? Please specify version as well if possible:
    [x] Windows
    [] MacOS
    [] Ubuntu
    [] Other (Please Specify)

**Comment by sadukie** [comment]
I just tried this on my other Azure Notebooks account, and while I can access pre-existing notebooks, uploading this particular file presents the same problem.  So I wonder if it's something with this notebook file...

**Comment by sadukie** [comment]
[Python for C# Devs.zip](https://github.com/Microsoft/AzureNotebooks/files/2071261/Python.for.C.Devs.zip)

I've zipped the file so that you can see specific details for the ipynb file.

**Comment by smortaz** [comment]
Ok thanks. Does this happen for all your libraries or just this one?  Ie if
you create a library and then an empty notebook and right click / run in
jupyter lab, what’s the result?

And if you don’t select run in jupyter lab and just click the notebook ,
all is ok?

**Comment by smortaz** [comment]
Aha, that’s what I needed. Ok will take a look. Currently on phone...

What’s the url to the original library?

**Comment by sadukie** [comment]
The original ipynb file is attached in the ZIP file.  I created it locally via Jupyter Notebook on Windows 10.  The original Azure Notebook library I was trying to use was for my cletechconsulting account.

**Comment by sadukie** [comment]
Locally, I ran this:

```
import sys
sys.version
```
Locally, I am running:
'3.6.4 |Anaconda custom (64-bit)| (default, Jan 16 2018, 10:22:32) [MSC v.1900 64 bit (AMD64)]'

**Comment by smortaz** [comment]
I understand. But can you share the URL for your Library that I can clone?
  I’d like to start on azure notebooks and clone versus creating one from
your zip file

**Comment by sadukie** [comment]
Sure!  The library is at: https://notebooks.azure.com/cletechconsulting/libraries/introtopyforcsharpdevs

**Comment by sadukie** [comment]
I found my problem... once I removed the "#" from the file name, it was happier.  I tried this with my other account (sadukie).  So I'm sure that once I update cletechconsulting's file by removing the #, I should be fixed on my end.

**Comment by smortaz** [comment]
Ok great. Either way it’s a bug we need to fix. Thanks!

**Comment by sadukie** [comment]
Confirmed that this is fixed with cletechconsulting too by renaming the file: https://introtopyforcsharpdevs-cletechconsulting.notebooks.azure.com/nb/notebooks/Python%20for%20Devs.ipynb

What tipped me off was the redirect.   I used Fiddler 4 to sniff the traffic to see just how the network calls were getting made.  When I looked at the URL and the sniffed traffic logs, I suspected something was wrong related to the filename.
